### PR TITLE
Blocking :wq

### PR DESCRIPTION
**Pull request by kirawi**

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -1203,15 +1202,35 @@ mod cmd {
...
-            return;
+        match write_impl(view, doc, args.first()) {
+            Ok(handle) => {
+                if let Err(e) = helix_lsp::block_on(handle) {
+                    cx.editor.set_error(e.to_string());
+                    return;
+                }
+            }
+            Err(e) => {
+                cx.editor.set_error(e.to_string());
```
Comment (pickfire) [inline helix-term/src/commands.rs:1213]: Can we even see the error when we already quit?

Comment (pickfire) [reply helix-term/src/commands.rs:1213]: Ah, nevermind. Yeah, it returns here.

Comment (kirawi) [reply helix-term/src/commands.rs:1213]: It should return an error before it quits.

File: helix-term/src/commands.rs [hunk 2]
```diff
@@ -1203,15 +1202,35 @@ mod cmd {
...
+                return;
+            }
         };
         quit(cx, &[], event)
```
Comment (pickfire) [inline helix-term/src/commands.rs:1217]:
```suggestion
        match write_impl(view, doc, args.first()) {
            Ok(handle) => {
                if let Err(e) = helix_lsp::block_on(handle) {
                    cx.editor.set_error(e.to_string());
                } else {
                    quit(cx, &[], event);
                }
            }
            Err(e) => {
                cx.editor.set_error(e.to_string());
            }
        };
```

File: helix-term/src/commands.rs [hunk 3]
```diff
@@ -1203,15 +1202,34 @@ mod cmd {
...
+                    cx.editor.set_error(e.to_string());
+                    return;
+                }
+            }
+            Err(e) => {
+                cx.editor.set_error(e.to_string());
+                return;
+            }
+        };
         force_quit(cx, &[], event);
```
Comment (pickfire) [inline helix-term/src/commands.rs:1233]:
```suggestion
        let (view, doc) = current!(cx.editor);
        match write_impl(view, doc, args.first()) {
            Ok(handle) => {
                if let Err(e) = helix_lsp::block_on(handle) {
                    cx.editor.set_error(e.to_string());
                } else {
                    force_quit(cx, &[], event);
                }
            }
            Err(e) => {
                cx.editor.set_error(e.to_string());
            }
        }
```

**Review by pickfire — APPROVED** [review]
Looks good to me. Tested and works. We might need an async variant later.

**Review by archseer — APPROVED** [review]

**Pull request merged**